geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala

```diff
@@ -586,6 +655,64 @@ object GeoMesaDataStore extends LazyLogging {
     }
   }
 
+  sealed trait SchemaCompatibility {
+
+    /**
+     * Ensures that the schema matches the existing schema in the data store, or throws an error if
+     * the schemas are incompatible
+     */
+    def apply(): Unit
+  }
+
+  object SchemaCompatibility {
+
+    /**
+     * Indicates that the schema is equal to the existing schema in the data store
+     */
+    case object Unchanged extends SchemaCompatibility {
+      override def apply(): Unit = {}
+    }
+
+    /**
+     * Indicates that the schema does not exist in the data store
+     *
+     * @param ds data store
+     * @param sft schema
+     */
+    class Missing(ds: GeoMesaDataStore[_], val sft: SimpleFeatureType) extends SchemaCompatibility {
```

Comment:
Would calling this something like "NewSchema" be ok with you? 

When I was reading the unit test, I thought it was something about missing data.

Reply:
how about MissingSchema or DoesNotExist? this is supposed to represent a status, so NewSchema is more of the action you'd take than the status.